pinot-transport/src/main/java/com/linkedin/pinot/requestHandler/MultipleOrEqualitiesToInClauseFilterQueryTreeOptimizer.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.requestHandler;
+
+import com.google.common.base.Splitter;
+import com.linkedin.pinot.common.request.FilterOperator;
+import com.linkedin.pinot.common.utils.StringUtil;
+import com.linkedin.pinot.common.utils.request.FilterQueryTree;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+
+/**
+ * Optimizer that collapses multiple OR clauses to IN clauses. For example, <code>a = 1 OR a = 2 OR a =
+ * 3</code> gets turned to <code>a IN (1, 2, 3)</code>.
+ */
+public class MultipleOrEqualitiesToInClauseFilterQueryTreeOptimizer implements FilterQueryTreeOptimizer {
+  @Override
+  public FilterQueryTree optimize(FilterQueryTree filterQueryTree) {
+    return optimize(filterQueryTree, null);
+  }
+
+  private FilterQueryTree optimize(FilterQueryTree filterQueryTree, FilterQueryTree parent) {
+    if (filterQueryTree.getOperator() == FilterOperator.OR) {
```

Comment:
- break this method into multiple methods to improve readability.
- it looks like we always rewrite the filterTree even when there is no optimization possible